### PR TITLE
(fix) colima: use a docker context specific to runner; prevent duplicate start

### PR DESCRIPTION
**Pull request by tobitege**
**What is the problem that this fixes or functionality that this introduces? Does it fix any open issues?**

This PR attempts to fix colima startup by 
- preventing to start colima 2 times in a row (first with brew, then in `for` loop)
- use a docker context to separate docker resources for the runner instance
- use a random port number

---

**Give a summary of what the PR does, explaining any non-trivial design decisions**

- removed the initial mangling of colima start and delete, which shouldn't be necessary
- added env variable DOCKER_CONTEXT to be used by colima

### CODE REVIEW

**Comment by tobitege** [comment]
Doh! --context isn't supported. Closing again. 🤣

**Review by neubig — APPROVED** [review]
Looks great to me, and it's passing tests! Let's try this.

**Comment by SmartManoj** [comment]
[Again failed](https://github.com/OpenDevin/OpenDevin/actions/runs/10087360615/job/27891383693#step:6:367)

File: .github/workflows/run-unit-tests.yml [hunk 1]
```diff
@@ -83,16 +83,23 @@ jobs:
...
+          docker context create $DOCKER_CONTEXT --docker host=unix:///var/run/docker.sock
 
           start_colima() {
-            colima start --network-address --arch x86_64 --cpu=1 --memory=1
+            # Find a free port
+            RANDOM_PORT=$((RANDOM % 16384 + 49152))
+
+            echo "Using random port: $RANDOM_PORT for SSH"
+
+            colima start --network-address --arch x86_64 --cpu=1 --memory=1 --ssh-port $RANDOM_PORT
```
Comment (SmartManoj) [inline .github/workflows/run-unit-tests.yml:97]: is ssh-port a valid arg?
https://github.com/abiosoft/colima/blob/c8c4c5a69b4e422dab76ac0a0f81515094302c2e/cmd/start.go#L148-L156

File: .github/workflows/run-unit-tests.yml [hunk 2]
```diff
@@ -83,16 +83,23 @@ jobs:
...
+            docker context use $DOCKER_CONTEXT
```
[the remark this answers is not in the document]
Comment (tobitege) [reply .github/workflows/run-unit-tests.yml:98]: Ask your AI 😃

[the remark this answers is not in the document]
Comment (SmartManoj) [reply .github/workflows/run-unit-tests.yml:98]: https://www.perplexity.ai/search/use-of-docker-context-with-loc-p7vSyOqZSlO9FjCjwfyX0A

[the remark this answers is not in the document]
Comment (tobitege) [reply .github/workflows/run-unit-tests.yml:98]: And still I think there are more benefits to using it than not.

[the remark this answers is not in the document]
Comment (SmartManoj) [reply .github/workflows/run-unit-tests.yml:98]: Also. this line changes the exit status of this function. Source: [`man bash`](https://www.man7.org/linux/man-pages/man1/bash.1.html#:~:text=When%20executed%2C%20the%0A%20%20%20%20%20%20%20%20%20%20%20%20%20%20exit%20status%20of%20a%20function%20is%20the%20exit%20status%20of%20the%20last%0A%20%20%20%20%20%20%20%20%20%20%20%20%20%20command%20executed%20in%20the%20body.)

Comment (SmartManoj) [inline .github/workflows/run-unit-tests.yml:98]: Also. this line changes the exit status of this function. Source: [man bash](https://www.man7.org/linux/man-pages/man1/bash.1.html#:~:text=When%20executed%2C%20the%0A%20%20%20%20%20%20%20%20%20%20%20%20%20%20exit%20status%20of%20a%20function%20is%20the%20exit%20status%20of%20the%20last%0A%20%20%20%20%20%20%20%20%20%20%20%20%20%20command%20executed%20in%20the%20body.)

[the remark this answers is not in the document]
Comment (tobitege) [reply .github/workflows/run-unit-tests.yml:98]: That I'll look into later today.

[the remark this answers is not in the document]
Comment (SmartManoj) [reply .github/workflows/run-unit-tests.yml:98]: is there any difference if the last two lines are swapped?

File: .github/workflows/run-unit-tests.yml [hunk 3]
```diff
@@ -83,16 +83,23 @@ jobs:
           fi
           rm -rf ~/.colima ~/.lima
           brew install --HEAD colima
-          brew services start colima
```
Comment (SmartManoj) [inline .github/workflows/run-unit-tests.yml:86]: https://github.com/abiosoft/colima/issues/960

**Pull request merged**